src/pybind/mgr/orchestrator/module.py

```diff
@@ -473,103 +477,214 @@ def set_unmanaged_flag(self, service_name: str, unmanaged_flag: bool) -> HandleC
             return HandleCommandResult(stdout=f"No specs found with the <service_name> -> {service_name}")
 
     @_cli_write_command(
-        'orch apply osd',
-        'name=all_available_devices,type=CephBool,req=false '
-        'name=preview,type=CephBool,req=false '
+        'orch osd spec',
         'name=service_name,type=CephString,req=false '
+        'name=preview,type=CephBool,req=false '
         'name=unmanaged,type=CephBool,req=false '
         "name=format,type=CephChoices,strings=plain|json|json-pretty|yaml,req=false",
-        'Create OSD daemon(s) using a drive group spec')
-    def _apply_osd(self,
-                   all_available_devices: bool = False,
-                   preview: bool = False,
-                   service_name: Optional[str] = None,
-                   unmanaged: bool = False,
-                   format: Optional[str] = 'plain',
-                   inbuf: Optional[str] = None) -> HandleCommandResult:
-        """Apply DriveGroupSpecs to create OSDs"""
+        'Common operations on an OSDSpec. Allows previewing and changing the unmanaged flag.')
```

Comment:
which is what `apply` should do. i think we need another iteration of cli cleanups.

Reply:
I think `apply` should actually only _apply_.

There is an entire maintenance part that should be somehow separated. If `apply` is the main entrypoint of an command there is not much room for subcommands that make sense.

But I agree, we need to have a proper discussion on how to proceed with the CLI development.

Reply:
> But I agree, we need to have a proper discussion on how to proceed with the CLI development.

:+1: